### PR TITLE
fixing parameter for gateway

### PR DESCRIPTION
**Pull request by MahendraAgrawal**
Fixing dnslable and IPAddressName for gateway.

### CODE REVIEW

File: rds-deployment/azuredeploy.json [hunk 1]
```diff
@@ -2,12 +2,19 @@
...
   "parameters": {
-    "dnsLabelPrefix": {
+    "gwdnsLabelPrefix": {
       "type": "string",
       "metadata": {
-        "description": "Unique public DNS prefix for the deployment. The fqdn will look something like '<dnsname>.westus.cloudapp.azure.com'. Up to 62 chars, digits or dashes, lowercase, should start with a letter: must conform to '^[a-z][a-z0-9-]{1,61}[a-z0-9]$'. For example johndns1 will result the final RDWEB access url like https://johndns1.westus.cloudapp.azure.com/RDWeb"
+        "description": "Unique gateway public DNS prefix for the deployment. The fqdn will look something like '<dnsname>.westus.cloudapp.azure.com'. Up to 62 chars, digits or dashes, lowercase, should start with a letter: must conform to '^[a-z][a-z0-9-]{1,61}[a-z0-9]$'. For example johndns1 will result the final RDWEB access url like https://johndns1.westus.cloudapp.azure.com/RDWeb"
       }
     },
+    "gwpublicIPAddressName": {
```
Comment (bmoore-msft) [inline rds-deployment/azuredeploy.json]: move this to a variable

File: rds-deployment/azuredeploy.json [hunk 2]
```diff
@@ -102,24 +108,23 @@
...
     "gwlbID": "[resourceId('Microsoft.Network/loadBalancers',variables('gwLBName'))]",
     "gwlbFEConfigID": "[concat(variables('gwlbID'),'/frontendIPConfigurations/',variables('gwlbFE'))]",
     "gwlbBE": "GWLBBE",
     "gwBEAddressPoolID": "[concat(variables('gwlbID'),'/backendAddressPools/',variables('gwlbBE'))]",
-    "gwdnsLabelPrefix": "[concat('gwd', resourceGroup().name)]",
+    "dnsLabelPrefix": "[tolower(concat('adns', resourceGroup().name))]",
     "storageAccountName": "[tolower(concat('rdsa',uniqueString(resourceGroup().id)))]",
     "storageAccountType": "Standard_LRS",
-    "uniqueStorageAccountContainerName": "[concat('sc', resourceGroup().name)]",
+    "uniqueStorageAccountContainerName": "[tolower(concat('sc', resourceGroup().name))]",
```
Comment (bmoore-msft) [inline rds-deployment/azuredeploy.json]: resourcegroup names can contain chars that are not valid for storage accounts, better practice would be to use the uniqueString() function

**Comment by MahendraAgrawal** [comment]
Hi Brian, I have fixed the container name to use uniquestring; however, gwpublicIPAddress name is required as parameter for scenario where partners/users will provide the IPAddress name they want to identify their gateway. It was variable before and after feedback from them, we made this parameter in this submission. Please help merge this pull request. Thanks,
Mahendra.

**Comment by bmoore-msft** [comment]
I think it makes sense to prompt for DNS in this case, but the name of the PIP resource, that's probably not necessary - you could hardcode that in a variable, it's not end-user facing.

**Comment by MahendraAgrawal** [comment]
Looping Christian as he is our PM and mentioned for this requirement. He has also developed a solution template based on it.

**Comment by MahendraAgrawal** [comment]
Yes, this is so the template can also be the basis for the Azure Marketplace Solutions Template. In order for the Solutions Template to correctly validate that the DNS name is valid and available to use from Azure, we need to use the PublicIPAddressComboxBox item, which will ask for a name for the PIP resource. We want to make sure that the name of the resource being created matches what they entered, hence why there’s  parameter for it.

Thanks

From: Mahendra Agrawal
Sent: Monday, November 7, 2016 8:23 AM
To: Azure/azure-quickstart-templates <reply@reply.github.com>; Azure/azure-quickstart-templates <azure-quickstart-templates@noreply.github.com>
Cc: Author <author@noreply.github.com>; Christian Montoya <chrimo@microsoft.com>
Subject: RE: [Azure/azure-quickstart-templates] fixing parameter for gateway (#2712)

Looping Christian as he is our PM and mentioned for this requirement. He has also developed a solution template based on it.

**Pull request merged**